### PR TITLE
Tickets/dm 13896

### PR DESCRIPTION
**Pull request by fritzm**

### CODE REVIEW

File: LDM-135.tex [hunk 1]
```diff
@@ -1764,16 +1764,23 @@ \subsection{Fault Tolerance}\label{fault-tolerance}
...
-their analysis algorithms in a procedural language, such as Python. To do
-that, we will allow users to run their own code on their own hardware
-resources co-located with production database servers. Users then run queries
-on the production database which stream rows directly from database cluster
-nodes to the user processing cluster, where arbitrary code may run without
-endangering the production database. This allows their incurred database I/O
-needs to be satisfied using the database system's shared scanning
-infrastructure while providing the full flexibility of running arbitrary code.
+express through SQL language; e.g. time-series analyses. For these types of
+analyses, we wish to allow users to execute their algorithms in a procedural
```
Comment (ktlim) [inline LDM-135.tex:1768]: "analyses" -> "analysis".

File: LDM-135.tex [hunk 2]
```diff
@@ -1764,16 +1764,23 @@ \subsection{Fault Tolerance}\label{fault-tolerance}
...
+language, such as Python, on hardware resources co-located with the production
+database servers.
```
Comment (ktlim) [inline LDM-135.tex]: Might be slightly better to say "production database _data_" rather than "servers".

File: LDM-135.tex [hunk 3]
```diff
@@ -1764,16 +1764,23 @@ \subsection{Fault Tolerance}\label{fault-tolerance}
...
+
+One approach being investigated is to allow users to run pre-filter queries on
+the production database, with the results streamed to the user processing cluster
+where their custom code may run without directly endangering the production database.
+This would allow incurred database I/O needs to be satisfied using the database
+system's shared scanning infrastructure.
+
+Another approach being investigated would involve keeping one set of replicas of
+the catalog data files external to the the database system, in a storage format
```
Comment (ktlim) [inline LDM-135.tex]: "the the"

File: LDM-135.tex [hunk 4]
```diff
@@ -1764,16 +1764,23 @@ \subsection{Fault Tolerance}\label{fault-tolerance}
...
-needs to be satisfied using the database system's shared scanning
-infrastructure while providing the full flexibility of running arbitrary code.
+express through SQL language; e.g. time-series analyses. For these types of
+analyses, we wish to allow users to execute their algorithms in a procedural
+language, such as Python, on hardware resources co-located with the production
+database servers.
+
+One approach being investigated is to allow users to run pre-filter queries on
+the production database, with the results streamed to the user processing cluster
+where their custom code may run without directly endangering the production database.
```
Comment (ktlim) [inline LDM-135.tex]: This begs a lot of questions about API, setting up multiple parallel connections, synchronization/buffering/dropping, etc., but I guess they're out of scope for this change.

**Review by ktlim — APPROVED** [review]

File: LDM-135.tex (hunk 1 above)
Comment (fritzm) [reply LDM-135.tex:1768]: Both usages above are intended to be plural, though?

**Pull request merged**